searchlib/src/vespa/searchlib/features/attributefeature.cpp

```diff
@@ -365,6 +379,12 @@ createAttributeExecutor(const IAttributeVector *attribute, const vespalib::strin
             { SingleValueExecutorCreator<IntegerAttributeTemplate<int8_t>> creator;       if (creator.handle(attribute)) return creator.create(stash); }
             { SingleValueExecutorCreator<IntegerAttributeTemplate<int32_t>> creator;      if (creator.handle(attribute)) return creator.create(stash); }
             { SingleValueExecutorCreator<IntegerAttributeTemplate<int64_t>> creator;      if (creator.handle(attribute)) return creator.create(stash); }
+            {
+                auto boolAttribute = dynamic_cast<const SingleBoolAttribute *>(attribute);
+                if (boolAttribute) {
+                    return stash.create<BoolAttributeExecutor>(*boolAttribute);
```

Comment:
This should be possible if and only if we also chose to drop additional outputs.
It feels a bit fragile to check meta-data in one end and actual attribute implementation class in the other end.

Reply:
Fixed and added some more sanity checks.